web/packages/core/src/ruffle-player.ts

```diff
@@ -638,9 +638,18 @@ export class RufflePlayer extends HTMLElement {
                 this.instance!.stream_from(this.swfUrl.href, parameters);
             } else if ("data" in options) {
                 console.log("Loading SWF data");
+
+                const movieName = swfFileName(
+                    new URL(
+                        options.swfFileName || "movie.swf",
+                        document.baseURI
+                    )
+                );
+
                 this.instance!.load_data(
                     new Uint8Array(options.data),
-                    sanitizeParameters(options.parameters)
+                    sanitizeParameters(options.parameters),
+                    new URL(movieName, document.baseURI).href
```

Comment:
We should do this in WASM and not JS, just to be safe. If the WASM side asks for the URL from JS, it's a simple JS script to call it with anything else (or even just search-and-replace for the lazy malicious host)

Reply:
I don't really know how to do that from WASM, but feel free to push that change to my PR